light-system-programs/programs/merkle_tree_program/src/lib.rs

```diff
@@ -55,10 +55,12 @@ pub mod merkle_tree_program {
             return err!(ErrorCode::InvalidAuthority);
         }
 
-        if ctx.remaining_accounts.len() != 1 {
-            return err!(ErrorCode::ExpectedOldMerkleTree);
+        if ctx.remaining_accounts.len() != 2 {
+            return err!(ErrorCode::ExpectedOldMerkleTrees);
```

Comment:
I was briefly wondering whether we should add checks here that the correct accounts are being passed but we check that when we de-serialize already right?

Reply:
Yeah, I think that the deserialization, which happens later is aleady doing the job?